scripts/setStages.ts

```diff
@@ -24,6 +24,31 @@ interface StageConfig {
   variableWalletLimitPath?: string;
 }
 
+function isValidDate(dateString: string): boolean {
+  const date = new Date(dateString);
```

Comment:
https://stackoverflow.com/questions/1214234/javascript-date-parsing-bug-fails-for-dates-in-june

Should be good based on this article, defining <dateString> and using that will fix

Reply:
<img width="1492" alt="image" src="https://github.com/me-foundation/erc721m/assets/95395274/6a6adc13-f556-4ac5-9b7b-685b6dcf5bdd">
Seems `new Date()` will not fail

Reply:
@FwazB I have suffered immensely from JS dates. It's a "whack a mole" problem. There's always another gotcha.

I am seeing seconds (from block.timestamp) used and date strings like @channing-magiceden is calling out. Something like this could be helpful, modify as necessary:

``` typescript
function isValidDate(date: string) {
  if (/^\d+$/.test(date)) {
    // Date expects milliseconds, * 1000 here if using seconds
    const milliSeconds = parseInt(date, 10);

    return isValidDateInMilliSeconds(milliSeconds);
  } else {
    return isValidDateString(date);
  }
}

function isValidDateInMilliSeconds(milliseconds: number) {
  return !isNaN(new Date(milliseconds).getDate());
}

function isValidDateString(dateString: string) {
  const [year, month, day] = dateString.split(/[-/]/).map(Number);
  const date = new Date(year, month - 1, day);

  if (
    !(
      date.getFullYear() === year &&
      date.getMonth() + 1 === month &&
      date.getDate() === day
    )
  ) {
    return false;
  } else {
    return !isNaN(date.getTime());
  }
}

```

But this has some things to be aware of as well:

`isValidDate("20240631") === true`because the regex will treat this as milliseconds. 

But in general, with js dates you can split/compare the input against the date values to ensure they match.